src/Components/Artwork/useArtworkLists.ts

```diff
@@ -28,18 +30,37 @@ export const useArtworkLists = (options: SaveArtworkToListsOptions) => {
       return
     }
 
-    if (artwork.isSavedToCustomLists) {
-      // Display select lists for artwork modal if artwork is saved to the custom lists
-      openSelectListsForArtworkModal()
-    } else if (artwork.isSavedToDefaultList) {
-      // Display toast if artwork is already saved to the default list
-      showToastByAction(ResultAction.SavedToDefaultList, artwork.isInAuction)
-    } else {
-      // Save artwork to the default list
-      saveArtworkToLists()
+    const fetchSavedStatusAndPerformAction = async () => {
+      const data = await fetchQuery<useArtworkListsArtworkSaveStatesQuery>(
+        relayEnvironment,
+        graphql`
+          query useArtworkListsArtworkSaveStatesQuery($artworkID: String!) {
+            artwork(id: $artworkID) {
+              isSavedToList
+              isSaved
+            }
+          }
+        `,
+        { artworkID: artwork.internalID }
+      ).toPromise()
+
+      // The user clicked the save button while logged out.
+      // We need to comprehensively check the artwork's save state.
```

Comment:
This is analogous to the code change above (checking list inclusion on 'unsave'). This branch is for when someone clicks an (unsaved) save button while logged out - then they login - what action to take after login.

The artwork in question might be saved by them already to a custom list, _just_ to their default saves, or not saved at all.

So similarly, we want to comprehensively fetch the current saved state to direct them properly.